src/test/rpc/LedgerRPC_test.cpp

```diff
@@ -0,0 +1,283 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/protocol/ErrorCodes.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/test/jtx.h>
+#include <ripple/beast/unit_test.h>
+
+namespace ripple {
+
+class LedgerRPC_test : public beast::unit_test::suite
+{
+    static
+    std::unique_ptr<Config>
+    makeNonAdminConfig()
+    {
+        auto p = std::make_unique<Config>();
+        test::setupConfigForUnitTests(*p);
+        (*p)["port_rpc"].set("admin","");
+        (*p)["port_ws"].set("admin","");
+        return p;
+    }
+
+public:
+    void testBadInput()
+    {
+        testcase("Bad Input");
+        using namespace test::jtx;
+        Env env { *this };
+        Account const gw { "gateway" };
+        auto const USD = gw["USD"];
+        Account const bob { "bob" };
+
+        env.fund(XRP(10000), gw, bob);
+        env.trust(USD(1000), bob);
+        env.close();
+
+        {
+            Json::Value jvParams;
+            jvParams[jss::ledger_index] = "0"; // NOT an integer
+            auto const jrr = env.rpc ( "json", "ledger", to_string(jvParams) ) [jss::result];
```

Comment:
I get what you are saying, but my observation (so far) has been that if "result" is not there, then we are so far sideways that it doesn't matter. The RPC Client invoke also seems to be assuming "result" is always there:

https://github.com/ripple/rippled/blob/2fa5358254924f2f70f191b14357c3e33197403b/src/ripple/test/impl/JSONRPCClient.cpp#L134

...and it's also not checking the return status of parse (bummer!). 

So my inclination is to leave this as is (I don't tons of value in checking for existing of a field that the invoke-er has already assumed is there..) and I like that we don't create lots of noise in the code making low-value checks/asserts.

All that said, if the consensus is to always check for existence of result, then I will gladly change the code. Another option is - perhaps - to add the check to the rpc invocation chain somewhere.